pdns/dnsdist-protobuf.cc

```diff
@@ -149,13 +149,21 @@ void DNSDistProtoBufMessage::serialize(std::string& data) const
   }
   else if (distProto == dnsdist::Protocol::DoH) {
     protocol = pdns::ProtoZero::Message::TransportProtocol::DoH;
+    m.setHTTPVersion(pdns::ProtoZero::Message::HTTPVersion::HTTP2);
```

Comment:
Technically it could be HTTP/1.1, when DoH is provided by h2o, but we unfortunately don't expose the HTTP version at the moment. I vote for ignoring this since we will be deprecating h2o support soon anyway.

Reply:
Wouldn't something like this do the trick ? also the `HAVE_LIBH2OEVLOOP` define would make it easy to find when removing the legacy code.

```suggestion
+#ifdef HAVE_DNS_OVER_HTTPS                                                                                                                                                                                                                                          #ifdef HAVE_LIBH2OEVLOOP
+    m.setHTTPVersion(pdns::ProtoZero::Message::HTTPVersion::HTTP1);
+#else
     m.setHTTPVersion(pdns::ProtoZero::Message::HTTPVersion::HTTP2);
+#endif
+#endif

Reply:
It's not that easy, I'm afraid. When `HAVE_LIBH2OEVLOOP` is not enabled it is indeed always HTTP/2, but otherwise it might be HTTP/1 or HTTP/2, and right now we don't pass this information to the protobuf/dnstap layer.

Reply:
We could add a method to `DNSQuestion` that would query a method in `DOHUnit` looking at `req->version`, but I'm not sure it's worth the complexity.